engine/benchmarks/src/main/scala/pl/touk/nussknacker/engine/benchmarks/interpreter/OneParamInterpreterBenchmark.scala

```diff
@@ -23,34 +25,66 @@ class OneParamInterpreterBenchmark {
     .id("t1")
     .exceptionHandlerNoParams()
     .source("source", "source")
+    .buildSimpleVariable("v1", "v1", "{a:'', b: 2}")
     .enricher("e1", "out", "service", "p1" -> "''")
+    //Uncomment to assess impact of costly variables
+    //.buildSimpleVariable("v2", "v2", "{a:'', b: #out, c: {'d','d','ss','aa'}.?[#this.substring(0, 1) == ''] }")
+    //.buildSimpleVariable("v3", "v3", "{a:'', b: #out, c: {'d','d','ss','aa'}.?[#this.substring(0, 1) == ''] }")
     .sink("sink", "#out", "sink")
-  private val interpreter = new InterpreterSetup[String].sourceInterpretation(process, Map("service" -> OneParamService), Nil)
+
+  private val realAsync = true
+
+  private val service = new OneParamService(realAsync)
+
+  private val interpreterFuture = new InterpreterSetup[String]
+    .sourceInterpretation[Future](process, Map("service" -> service), Nil)(new FutureShape()(SynchronousExecutionContext.ctx))
+
+  private val interpreterIO = new InterpreterSetup[String]
+    .sourceInterpretation[IO](process, Map("service" -> service), Nil)
+
+  private val interpreterFutureAsync = new InterpreterSetup[String]
+    .sourceInterpretation[Future](process, Map("service" -> service), Nil)(new FutureShape()(ExecutionContext.global))
 
 
   @Benchmark
   @BenchmarkMode(Array(Mode.Throughput))
   @OutputTimeUnit(TimeUnit.SECONDS)
-  def benchmarkSync(): AnyRef = {
-    Await.result(interpreter(Context(""), SynchronousExecutionContext.ctx), 1 second)
+  def benchmarkFutureSync(): AnyRef = {
+    Await.result(interpreterFuture(Context(""), SynchronousExecutionContext.ctx), 1 second)
   }
 
 
   @Benchmark
   @BenchmarkMode(Array(Mode.Throughput))
   @OutputTimeUnit(TimeUnit.SECONDS)
-  def benchmarkAsync(): AnyRef = {
-    Await.result(interpreter(Context(""), ExecutionContext.Implicits.global), 1 second)
+  def benchmarkFutureAsync(): AnyRef = {
+    Await.result(interpreterFutureAsync(Context(""), ExecutionContext.Implicits.global), 1 second)
+  }
+
+  @Benchmark
+  @BenchmarkMode(Array(Mode.Throughput))
+  @OutputTimeUnit(TimeUnit.SECONDS)
+  def benchmarkSyncIO(): AnyRef = {
+    interpreterIO(Context(""), SynchronousExecutionContext.ctx).unsafeRunSync()
+  }
+
+
+  @Benchmark
+  @BenchmarkMode(Array(Mode.Throughput))
+  @OutputTimeUnit(TimeUnit.SECONDS)
+  def benchmarkAsyncIO(): AnyRef = {
+    interpreterIO(Context(""), ExecutionContext.Implicits.global).unsafeRunSync()
   }
 
 
 
 }
-object OneParamService extends Service {
+class OneParamService(realAsync: Boolean) extends Service {
```

Comment:
Both are real ;) Maybe `instantlyCompletedFuture`? Like it is called in java api: https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/CompletableFuture.html#completedFuture-U-

Reply:
:+1: